modules/activemq-artemis-launch/added/drain.sh

```diff
@@ -1,177 +1,244 @@
+
   podName=$(echo $ENDPOINTS | python2 -c "import sys, json; print json.load(sys.stdin)['subsets'][0]['addresses'][${count}]['targetRef']['name']")
   if [ $? -ne 0 ]; then
-    echo "[drain.sh] Can't find pod name to scale down to tried ${count}"
-    exit 1
+    log "can't find pod name to scale down to tried ${count}"
+    continue
   fi
-  echo "[drain.sh] got podName ${podName} broker ip is ${BROKER_HOST}"
+
+  log "got endpoint pod name ${podName}"
   if [ "$podName" != "$BROKER_HOST" ]; then
-    # found an endpoint pod as a candidate for scaledown target
     podNamespace=$(echo $ENDPOINTS | python2 -c "import sys, json; print json.load(sys.stdin)['subsets'][0]['addresses'][${count}]['targetRef']['namespace']")
     if [ $? -ne 0 ]; then
-      echo "[drain.sh] Can't find pod namespace to scale down to tried ${count}"
-      exit 1
+      log "can't find pod namespace to scale down to tried ${count}"
+      continue
     fi
-    foundTarget="true"
-    break
-  fi
 
-  count=$(( count + 1 ))
-done
+    log "found an candidate target: $podName"
 
-if [ "$foundTarget" == "false" ]; then
-  echo "[drain.sh] Can't find a target to scale down to"
-  exit 1
-fi
+    IFSP=$IFS
+    IFS=
+    dnsNames=$(nslookup ${ip})
 
-# get host name of target pod
-IFSP=$IFS
-IFS=
-dnsNames=$(nslookup ${ip})
-echo "[drain.sh] $dnsNames"
-
-hostNamePrefix="${podName}.${HEADLESS_SVC_NAME}.${podNamespace}.svc."
-echo "[drain.sh] searching hostname with prefix: $hostNamePrefix"
-
-while read -r line
-do
-  IFS=' ' read -ra ARRAY <<< "$line"
-  if [ ${#ARRAY[@]} -gt 0 ]; then
-    hostName=${ARRAY[-1]}
-    if [[ $hostName == ${hostNamePrefix}* ]]; then
-      # remove the last dot
-      case $hostName in *.) hostName=${hostName%"."};; esac
-      echo "[drain.sh] found hostname: $hostName"
-      break
+    log "looked up dns entries $dnsNames"
+
+    hostNamePrefix="${podName}.${HEADLESS_SVC_NAME}.${podNamespace}.svc."
+
+    log "searching hostname with prefix: $hostNamePrefix"
+
+    while read -r line
+    do
+      IFS=' ' read -ra ARRAY <<< "$line"
+      if [ ${#ARRAY[@]} -gt 0 ]; then
+        hostName=${ARRAY[-1]}
+        if [[ $hostName == ${hostNamePrefix}* ]]; then
+          # remove the last dot
+          case $hostName in *.) hostName=${hostName%"."};; esac
+          log "found target hostname: $hostName"
+          break
+        fi
+      fi
+    done <<< ${dnsNames}
+    IFS=$IFSP
+
+    if [ -z "$hostName" ]; then
+      log "can't find target host name"
+      continue
     fi
-  fi
-done <<< ${dnsNames}
-IFS=$IFSP
 
-if [ -z "$hostName" ]; then
-  echo "[drain.sh] Can't find target host name"
-  exit 1
-fi
+    SCALE_TO_BROKER="${hostName}"
+    log "scale down target is: $SCALE_TO_BROKER"
 
-source /opt/amq/bin/launch.sh nostart
+    # Add connector to the pod to scale down to
+    log "removing any existing scaledownconnector"
+    sed -i "s/<connector name=\"scaledownconnector\">.*//" ${instanceDir}/etc/broker.xml
```

Comment:
another alternative is using the delete command, i.e.
```suggestion
    sed -i '/<connector name="scaledownconnector">.*/d' ${instanceDir}/etc/broker.xml
```

Reply:
ok, will use d command.